webclient/app/src/app/features/decision/DecisionOverviewMap.jsx

```diff
@@ -191,107 +244,307 @@ function DecisionOverviewMap() {
     ];
 
 
-    // Return the map component with minimum required styles
-    return (
-        <Box sx={{height: 'calc(100vh - 64px)', position: 'relative'}}>
-            <DeckGL
-                layers={layers}               // Add map layers
-                views={MAP_VIEW}              // Add map view settings
-                initialViewState={INITIAL_VIEW_STATE}  // Set initial position
-                controller={{dragRotate: false}}       // Disable rotation
-            />
-
-            <IconButton
-                onClick={() => setShowPanel(!showPanel)}
-                sx={{
-                    position: 'absolute',
-                    top: '10px',
-                    right: showPanel ? '320px' : '10px',
-                    bgcolor: 'white'
+
+    function DecisionDialog() {
```

Comment:
What was the reason to place the implementation of the DecisionDialog Component inside the file for the DecisionOverviewMap? Think about implementation options which would provide a better maintenance.